src/aap_eda/tasks/orchestrator.py

```diff
@@ -178,50 +178,61 @@ def dispatch(
+                    " with a queue.",
+                )
+            else:
+                LOGGER.info(
+                    f"Scheduling {process_parent_type} {process_parent_id} "
+                    f"to the least busy queue; its associated queue "
+                    "'{queue_name}' is from previous configuation settings.",
+                )
             queue_name = get_least_busy_queue_name()
-
         elif not check_rulebook_queue_health(queue_name):
-            msg = (
-                f"Queue {queue_name} has no workers. "
-                f"{process_parent_type} {process_parent_id} "
-                "is in unknown state. Waiting for readiness. "
-                "There might be a problem in the node, please contact "
-                "the administrator."
-            )
-            status_manager = StatusManager(
-                get_process_parent(process_parent_type, process_parent_id),
-            )
-            status_manager.set_status(
-                ActivationStatus.WORKERS_OFFLINE,
-                msg,
-            )
-            status_manager.set_latest_instance_status(
-                ActivationStatus.WORKERS_OFFLINE,
-                msg,
+            # The queue is unhealthy.  If we're not restarting it there's
+            # nothing we can do except update its status.
+            if request_type != ActivationRequest.RESTART:
+                msg = (
+                    f"{process_parent_type} {process_parent_id} is in an "
+                    "unknown state. The workers of its associated queue "
+                    f"'{queue_name}' are failing liveness checks. "
+                    "There may be an issue with the node; please contact "
+                    "the administrator."
+                )
+                status_manager = StatusManager(
+                    get_process_parent(process_parent_type, process_parent_id),
+                )
+                status_manager.set_status(
+                    ActivationStatus.WORKERS_OFFLINE,
+                    msg,
+                )
+                status_manager.set_latest_instance_status(
+                    ActivationStatus.WORKERS_OFFLINE,
+                    msg,
+                )
+                LOGGER.warning(msg)
+                return
+
+            # The queue is unhealthy, but this is a restart.
+            # The priority is to adhere to the restart policy and
+            # execute the task.
+            LOGGER.warning(
+                f"Restarting {process_parent_type} {process_parent_id} "
+                "on the least busy queue; The workers of its associated queue "
+                f"'{queue_name}' are failing liveness checks. "
+                "There may be an issue with the node; please contact "
+                "the administrator.",
             )
-            LOGGER.warning(msg)
-            return
+            queue_name = get_least_busy_queue_name()
```

Comment:
@jshimkus-rh This is the newly added line. Do we need it if the queue is checked to be unhealthy? or this is the fix: we continue for the `restart` type?

Reply:
This just a little refactor and it splits the message. There is around here another bug that will be addressed later. https://issues.redhat.com/browse/AAP-23378

Reply:
@hsong-rh Sorry, I didn't notice your question until Alejandro responded.

The functionality remains the same.  The flow got changed to have only one use of check_rulebook_queue_health which resulted in swapping the previous clauses with the restart case being the non-exceptional processing in the case of a non-healthy queue.